spl06_007.py

```diff
@@ -518,61 +522,64 @@ def _calculate_calibration_coefficients(self):
-        _c11_15_8 = self._i2c.read_register(SensorConstants.C11_15_8)
-        _c11_7_0 = self._i2c.read_register(SensorConstants.C11_7_0)
-        _c20_15_8 = self._i2c.read_register(SensorConstants.C20_15_8)
-        _c20_7_0 = self._i2c.read_register(SensorConstants.C20_7_0)
-        _c21_15_8 = self._i2c.read_register(SensorConstants.C21_15_8)
-        _c21_7_0 = self._i2c.read_register(SensorConstants.C21_7_0)
-        _c30_15_8 = self._i2c.read_register(SensorConstants.C30_15_8)
-        _c30_7_0 = self._i2c.read_register(SensorConstants.C30_7_0)
+        c[6] = self._i2c.read_register(SensorConstants.C10_15_8)
+        c[7] = self._i2c.read_register(SensorConstants.C10_7_0)
+        c[8] = self._i2c.read_register(SensorConstants.C01_15_8)
+        c[9] = self._i2c.read_register(SensorConstants.C01_7_0)
+        c[10] = self._i2c.read_register(SensorConstants.C11_15_8)
+        c[11] = self._i2c.read_register(SensorConstants.C11_7_0)
+        c[12] = self._i2c.read_register(SensorConstants.C20_15_8)
+        c[13] = self._i2c.read_register(SensorConstants.C20_7_0)
+        c[14] = self._i2c.read_register(SensorConstants.C21_15_8)
+        c[15] = self._i2c.read_register(SensorConstants.C21_7_0)
+        c[16] = self._i2c.read_register(SensorConstants.C30_15_8)
+        c[17] = self._i2c.read_register(SensorConstants.C30_7_0)
+        print('c = ',' '.join(map(str, c)))
 
         def most_significant_nibble(byte): return (byte & 0xf0) >> 4
 
         def least_significant_nibble(byte): return byte & 0x0f
 
-        c0 = self._twos_complement(
-            (_c0_11_4 << 4) | most_significant_nibble(_c0_3_0_c1_11_8),
-            12
-        )
-        c1 = self._twos_complement(
-            (least_significant_nibble(_c0_3_0_c1_11_8) << 8) | _c1_7_0,
-            12
-        )
-
-        c00 = self._twos_complement(
-            (_c00_19_12 << 12) | (_c00_11_4 << 4)
-            | most_significant_nibble(_c00_3_0_c10_19_16),
-            16
-        )
-        c10 = self._twos_complement(
-            (least_significant_nibble(_c00_3_0_c10_19_16) << 16)
-            | (_c10_15_8 << 8) | _c10_7_0,
-            16
-        )
-        c01 = self._twos_complement((_c01_15_8 << 8) | _c01_7_0, 16)
-        c11 = self._twos_complement((_c11_15_8 << 8) | _c11_7_0, 16)
-        c20 = self._twos_complement((_c20_15_8 << 8) | _c20_7_0, 16)
-        c21 = self._twos_complement((_c21_15_8 << 8) | _c21_7_0, 16)
-        c30 = self._twos_complement((_c30_15_8 << 8) | _c30_7_0, 16)
+        c0 = c[0] << 4 | c[1] >> 4; #working... returns 200
+
+        c1 = (c[1] & 0x0f) << 8 | c[2]
+        c1 = (-4096 | c1) if c1 & 1 << 11 else c1
+        print('c1=',c1)
+
+        c00 = c[3] << 12 | c[4] << 4 | c[5] >> 4;
+        #c00 = (0xfff00000 | c00) if (c00 & 1 << 19) else c00;
+        print('c00= ',c00)
+
+        c10 = ((c[5] & 0x0F) << 16) | (c[6] << 8) | c[7];
+        c10_temp = -1048576 if(c[5] & 0x8) else 0
```

Comment:
What's -1048576?

Reply:
yea I'll put a note in the code... -1048576 is negative hex for - 0x100000